### DISCUSSION TITLE
Add Suggestion Radio

### DISCUSSION THREAD

**Discussion by dorel14** [navidrome/navidrome#2799] [Ideas]
Hello 

I have on my computer a software named Mediamonkey wich helps me to organize my medias that have  a very good extension which is called LastFm Node ..
[http://www.mediamonkey.com/forum/viewtopic.php?t=24879](https://github.com/kagemomiji/airsonic-advanced/issues/url)

With this extension , it search on last fm for suggested song , look at my library and when it find a common song add it to my listening queue , so that i can listen for suggested song without having to choose them

it would be great having such a thing

**Decsription** 

A new entry in left  "listenbrainz radio" or "last fm radio" ,

I put 1 song in play queue , after i click on this button and it start searching for similar song in listenbrainz or lastfm and when found it looks at my medias , when something common is found the better choice it is added to play queue and while this song play ,  it search for another similarity  to add it in queue and so on ...

Top thing would be having the possibility to save this play queue as a playlist

**Comment by codeinabox** [comment]
ListenBrainz does have artist similarity endpoints that could be used as part of this, eg https://labs.api.listenbrainz.org/similar-artists?artist_mbids=584c04d2-4acc-491b-8a0a-e63133f4bfc4&algorithm=session_based_days_7500_session_300_contribution_5_threshold_10_limit_100_filter_True_skip_30

**Comment by kgarner7** [comment]
Also note, if you enable last.fm, you can use the [getsimilarsongs](https://opensubsonic.netlify.app/docs/endpoints/getsimilarsongs/) endpoint

**Comment by Batwam** [comment]
I've been searching for something like this forever. I'd love to have a radio or dynamic playlist implementing this. The reason I mention playlist is because not every mobile apps implements radios. 

I'm currently having to use Plexamp for this very feature. The next step which would be even neater for navidrome would be to implement something similar with a Sonic analysis. I don't know how difficult it would be to implement but I'm thinking that packages like immich seem to be able to implement AI models so, why not... just an idea.

**Comment by deluan** [comment]
You can create a Smart Playlist that can act as a radio. That's what I personally do, with a random selection of my favourites that I haven't listened in the past 90 days.

We are planning to add plugins next year, so hopefully we will be able to integrate thinsg like bliss-rs for sonic analysis: https://github.com/navidrome/navidrome/discussions/2259
